src/chart.rs

```diff
@@ -0,0 +1,91 @@
+pub struct Chart {
+    height: u32,
+    fill: char,
+    space: char,
+}
+
+impl Chart {
+    pub fn new() -> Chart {
+        Chart {
+            height: 10,
+            fill: '▌',
+            space: ' ',
+        }
+    }
+
+    pub fn height(mut self, h: u32) -> Chart {
+        self.height = h;
+        self
+    }
+
+    pub fn make<N>(&self, data: Vec<N>) -> String
+    where
+        N: Into<f64>,
+    {
+        let data: Vec<f64> = data.into_iter().map(|d| d.into()).collect();
+        let (min, max): (f64, f64) = data.iter().fold((0., 0.), |(min, max), datum| {
+            let datum = datum.clone();
+            (
+                if datum < min { datum } else { min },
+                if max < datum { datum } else { max },
+            )
+        });
+        let row_increment = (max - min) / self.height as f64;
+        let mut ret = String::with_capacity(self.height as usize * data.len());
+        for row in 0..self.height {
```

Comment:
could use a reduce function here?

Reply:
perhaps, not sure that improves readability.